scripts/esbuild/cli.ts

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs-extra';
+import { join } from 'path';
+import * as esbuild from 'esbuild';
+import type { BuildOptions as ESBuildOptions } from 'esbuild';
+import { CompilerOptions, nodeModuleNameResolver, parseConfigFileTextToJson, sys } from 'typescript';
+
+import { BuildOptions } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { getEsbuildAliases, getEsbuildExternalModules } from './util';
+
+/**
+ * Runs esbuild to bundle the `cli` submodule
+ *
+ * @param opts build options needed to generate the rollup configuration
+ * @returns an array containing the generated rollup options
+ */
+export async function buildCli(opts: BuildOptions) {
+  // clear out rollup stuff
+  fs.emptyDir(opts.output.cliDir)
+
+  const inputDir = join(opts.srcDir, 'cli');
+  const buildDir = join(opts.buildDir, 'cli');
+
+  const outputDir = opts.output.cliDir;
+  const esmFilename = join(outputDir, 'index.js');
+  const cjsFilename = join(outputDir, 'index.cjs');
+
+  const dtsFilename = 'index.d.ts';
+
+  const cliAliases = getEsbuildAliases(opts);
+
+  const external = getEsbuildExternalModules(opts, opts.output.cliDir);
+
+  const cliEsbuildOptions: ESBuildOptions = {
+    entryPoints: [join(inputDir, 'index.ts')],
```

Comment:
note that the entry point here is the typescript index file for the bundle, not the JS output by `tsc`